src/runtime_src/core/common/sensor.cpp

```diff
@@ -16,8 +13,248 @@
  * License for the specific language governing permissions and limitations
  * under the License.
  */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+#define XRT_CORE_COMMON_SOURCE
+
+// Local - Include Files
 #include "sensor.h"
-#include <boost/property_tree/json_parser.hpp>
+#include "query_requests.h"
+#include "utils.h"
+
+// System - Include Files
+#include <sstream>
+#include <iomanip>
+
+// ------ S T A T I C   F U N C T I O N S -------------------------------------
+namespace {
+/**
+ * Saves voltage-current pair of a sensor into a boost::property_tree
+ * Converts mV and mA into V and A before adding to the tree
+ * 
+ * @tparam QRVoltage voltage query for a sensor; query::noop if query DNE
+ * @tparam QRCurrent current query for the same sensor; query::noop if query DNE
+ * @param loc_id human readable sensor identifier
+ * @desc description about the sensor
+ */
+template <typename QRVoltage, typename QRCurrent>
+static boost::property_tree::ptree
+populate_sensor(const xrt_core::device * device, const std::string& loc_id, const std::string& desc)
+{
+  boost::property_tree::ptree pt;
+  pt.put("id", loc_id);
+  pt.put("description", desc);
+
+  uint64_t voltage = 0, current = 0;
+  try {
+    if (!std::is_same<QRVoltage, xrt_core::query::noop>::value)
+      voltage = xrt_core::device_query<QRVoltage>(device);
+  } catch (const std::exception& ex) {
+    pt.put("voltage.error_msg", ex.what());
+  }
+  pt.put("voltage.volts", xrt_core::utils::format_base10_shiftdown3(voltage));
+  pt.put("voltage.is_present", voltage != 0 ? "true" : "false");
+
+  try {
+    if (!std::is_same<QRCurrent, xrt_core::query::noop>::value)
+      current = xrt_core::device_query<QRCurrent>(device);
+  } catch (const std::exception& ex) {
+    pt.put("current.error_msg", ex.what());
+  }
+  pt.put("current.amps", xrt_core::utils::format_base10_shiftdown3(current));
+  pt.put("current.is_present", current != 0 ? "true" : "false");
+  
+  return pt;
+}
+
+template <typename QueryRequestType>
+static boost::property_tree::ptree
+populate_temp(const xrt_core::device * device, const std::string& loc_id, const std::string& desc)
+{
+  boost::property_tree::ptree pt;
+  uint64_t temp = 0;
```

Comment:
It maybe better to call this variable "temp_C" instead of temp.  For "temp" usually represents a "temporary" value.